src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/FileImportControl.java

```diff
@@ -93,6 +74,7 @@ protected boolean matchesExactly(String pkg, String fileName) {
             result = false;
         }
         else if (regex) {
+            final Pattern patternForExactMatch = createPatternForExactMatch(name);
```

Comment:
This is performance issue.
Compilation of pattern should be done ones, not on each try of match

Reply:
Done